src/praisonai-agents/praisonaiagents/memory/learn/stores.py

```
@@ -203,6 +234,85 @@ def clear(self) -> int:
+        """Append evicted entries to a recoverable archive (never hard-delete)."""
+        if not entries:
+            return
+        try:
+            import json
+            archive_path = self._archive_path()
+            existing: List[Dict[str, Any]] = []
+            if archive_path.exists():
+                try:
+                    with open(archive_path, "r", encoding="utf-8") as f:
+                        existing = json.load(f)
+                except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)